inference-engine/tests/functional/plugin/cpu/single_layer_tests/pooling.cpp

```diff
@@ -4,77 +4,115 @@
+        results << "Rounding=" << roundingType << "_";
+        results << "AutoPad=" << padType << "_";
+
+        results << CPUTestsBase::getTestCaseName(cpuParams);
+        results << CpuTestWithFusing::getTestCaseName(fusingParams);
+        return results.str();
+
+        return results.str();
     }
 
 protected:
     void SetUp() override {
-        poolLayerTestParamsSet basicParamsSet;
+        targetDevice = CommonTestUtils::DEVICE_CPU;
+
+        LayerTestsDefinitions::poolSpecificParams basicParamsSet;
+        PoolingInputShapes inputShapes;
+        InferenceEngine::Precision inPrc;
         CPUSpecificParams cpuParams;
         fusingSpecificParams fusingParams;
-        std::tie(basicParamsSet, cpuParams, fusingParams) = this->GetParam();
+        std::tie(basicParamsSet, inputShapes, inPrc, cpuParams, fusingParams) = this->GetParam();
 
-        poolSpecificParams poolParams;
-        std::vector<size_t> inputShape;
-        InferenceEngine::Precision netPrecision;
-        std::tie(poolParams, netPrecision, inPrc, outPrc, inLayout, outLayout, inputShape, targetDevice) = basicParamsSet;
-
-        if (outPrc == Precision::UNSPECIFIED) {
-            outPrc = inPrc;
-        }
+        ngraph::helpers::PoolingTypes poolType;
+        std::vector<size_t> kernel, stride;
+        std::vector<size_t> padBegin, padEnd;
+        ngraph::op::PadType padType;
+        ngraph::op::RoundingType roundingType;
+        bool excludePad;
+        std::tie(poolType, kernel, stride, padBegin, padEnd, roundingType, padType, excludePad) = basicParamsSet;
 
         std::tie(inFmts, outFmts, priority, selectedType) = cpuParams;
         std::tie(postOpMgrPtr, fusedOps) = fusingParams;
 
         if (selectedType.empty()) {
             selectedType = getPrimitiveType();
         }
-
         selectedType = selectedType + "_" + inPrc.name();
 
-        ngraph::helpers::PoolingTypes poolType;
-        std::vector<size_t> kernel, stride;
-        std::vector<size_t> padBegin, padEnd;
-        ngraph::op::PadType padType;
-        ngraph::op::RoundingType roundingType;
-        bool excludePad;
-        std::tie(poolType, kernel, stride, padBegin, padEnd, roundingType, padType, excludePad) = poolParams;
+        const auto ngPrc = FuncTestUtils::PrecisionUtils::convertIE2nGraphPrc(inPrc);
 
-        auto ngPrc = FuncTestUtils::PrecisionUtils::convertIE2nGraphPrc(netPrecision);
-        auto params = ngraph::builder::makeParams(ngPrc, {inputShape});
-        auto paramOuts = ngraph::helpers::convert2OutputVector(
-                ngraph::helpers::castOps2Nodes<ngraph::op::Parameter>(params));
+        if (!inputShapes.first.empty()) {
+            inputDynamicShapes = inputShapes.first;
+        } else {
+            inputDynamicShapes = {inputShapes.second.front()};
+        }
+        for (size_t i = 0; i < inputShapes.second.size(); i++) {
+            targetStaticShapes.push_back(std::vector<ngraph::Shape>{inputShapes.second[i]});
+        }
```

Comment:
Please use `init_input_shapes` call to initialize this arrays.

Reply:
`init_input_shapes` used fact that if `dynShape` scalar that means that we have static case
for me better to use empty vector
also `init_input_shapes` use `InputShape` as function parameter

Reply:
Again please refer to eltwise and softmax common test classes